deebot_client/commands/xml/common.py

```diff
@@ -76,11 +83,52 @@ def _handle_xml(cls, _: EventBus, xml: Element) -> HandlingResult:
             return HandlingResult.success()
 
         _LOGGER.warning(
-            'Command "%s" was not successful. XML response: %s', cls.NAME, xml
+            'Command "%s" was not successful. XML response: %s',
+            cls.NAME,
+            ElementTree.tostring(xml, "unicode"),
         )
         return HandlingResult(HandlingState.FAILED)
 
 
+class XmlCommandMqttP2P(XmlCommand, CommandMqttP2P, ABC):
+    """Json base command for mqtt p2p channel."""
+
+    @classmethod
+    def create_from_mqtt(cls, payload: str | bytes | bytearray) -> CommandMqttP2P:
+        """Create a command from the mqtt data."""
+        xml = ElementTree.fromstring(payload)
+        return cls._create_from_mqtt(xml.attrib)
+
+    @override
```

Comment:
```suggestion
    @final
    @override
```

So nobody will overwrite this function by mistake as `_handle_mqtt_p2p` is very similar